src/flamenco/vm/syscall/fd_vm_syscall_cpi.c

```diff
@@ -317,6 +317,11 @@ fd_vm_prepare_instruction( fd_instr_info_t const *  caller_instr,
   fd_borrowed_account_t * program_rec = NULL;
   int err = fd_txn_borrowed_account_view( instr_ctx->txn_ctx, &callee_instr->program_id_pubkey, &program_rec );
```

Comment:
Use `fd_txn_borrowed_account_view_idx` instead with `out_instr->program_id` to remove iteration.